### DISCUSSION TITLE
Batch queue ordering

### DISCUSSION THREAD

**Discussion by cblach** [jackc/pgx#1902] [Q&A] [answered]
Hello, I was wondering:
Are queries used in pgx batches guaranteed to be executed in the order which they are queued?

This is relevant when using locks:

```
b := pgx.Batch{}
b.Queue(someQuery)
b.Queue("SELECT pg_advisory_lock(1000)")
b.Queue(someInsertQuery)
br := conn.SendBatch(ctx, &b)
```

So that the lock is guaranteed to be executed before the insert query.

**Comment by jackc** [comment, marked as answer]
Yes.

**Comment by cblach** [comment]
Thank you for your clarification
